File: data-structure-tree/inorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> result = new LinkedList<Integer>();
        if (root == null){
            return result;
        }
        //check left
        if (root.left !=null) result.addAll(inorderTraversal(root.left));
        //check the root
        result.add(root.val);
        //check right
        if (root.right != null) result.addAll(inorderTraversal(root.right));
        return result;        
    }
}

File: queue-stack/inorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        if not root:
            return None
        self.result = []
        self.helper(root, self.result)
        return self.result

    def helper(self, root, result):
        if root:
            if root.left:
                self.helper(root.left, result)
            self.result.append(root.val)
            if root.right:
                self.helper(root.right, result)
